controllers/provisioners/eks/scaling/launchtemplate_test.go

```diff
@@ -0,0 +1,611 @@
+				LaunchTemplateData: &ec2.ResponseLaunchTemplateData{
+					IamInstanceProfile: &ec2.LaunchTemplateIamInstanceProfileSpecification{
+						Arn: aws.String(""),
+					},
+					SecurityGroupIds: aws.StringSlice([]string{}),
+					InstanceType:     aws.String("m5.xlarge"),
+				},
+			},
+			input: &CreateConfigurationInput{
+				SecurityGroups: []string{},
+				InstanceType:   "m5.large",
+			},
+			shouldDrift: true,
+		},
+		{
+			launchTemplate: &ec2.LaunchTemplate{
+				LaunchTemplateName: aws.String("my-launch-config"),
+			},
+			latestVersion: &ec2.LaunchTemplateVersion{
+				LaunchTemplateData: &ec2.ResponseLaunchTemplateData{
+					IamInstanceProfile: &ec2.LaunchTemplateIamInstanceProfileSpecification{
+						Arn: aws.String("arn::aws:my-role"),
+					},
+					SecurityGroupIds: aws.StringSlice([]string{}),
+				},
+			},
+			input: &CreateConfigurationInput{
+				SecurityGroups:        []string{},
+				IamInstanceProfileArn: "arn::aws:other-role",
+			},
+			shouldDrift: true,
+		},
+		{
+			launchTemplate: &ec2.LaunchTemplate{
+				LaunchTemplateName: aws.String("my-launch-config"),
+			},
+			latestVersion: &ec2.LaunchTemplateVersion{
+				LaunchTemplateData: &ec2.ResponseLaunchTemplateData{
+					IamInstanceProfile: &ec2.LaunchTemplateIamInstanceProfileSpecification{
+						Arn: aws.String(""),
+					},
+					SecurityGroupIds: aws.StringSlice([]string{"sg-1"}),
+				},
+			},
+			input: &CreateConfigurationInput{
+				SecurityGroups: []string{"sg-2"},
+			},
+			shouldDrift: true,
+		},
+		{
+			launchTemplate: &ec2.LaunchTemplate{
+				LaunchTemplateName: aws.String("my-launch-config"),
+			},
+			latestVersion: &ec2.LaunchTemplateVersion{
+				LaunchTemplateData: &ec2.ResponseLaunchTemplateData{
+					IamInstanceProfile: &ec2.LaunchTemplateIamInstanceProfileSpecification{
+						Arn: aws.String(""),
+					},
+					SecurityGroupIds: aws.StringSlice([]string{}),
+					KeyName:          aws.String("other-key"),
+				},
+			},
+			input: &CreateConfigurationInput{
+				SecurityGroups: []string{},
+				KeyName:        "my-key",
+			},
+			shouldDrift: true,
+		},
+		{
+			launchTemplate: &ec2.LaunchTemplate{
+				LaunchTemplateName: aws.String("my-launch-config"),
+			},
+			latestVersion: &ec2.LaunchTemplateVersion{
+				LaunchTemplateData: &ec2.ResponseLaunchTemplateData{
+					IamInstanceProfile: &ec2.LaunchTemplateIamInstanceProfileSpecification{
+						Arn: aws.String(""),
+					},
+					SecurityGroupIds: aws.StringSlice([]string{}),
+					UserData:         aws.String("hello"),
+				},
+			},
+			input: &CreateConfigurationInput{
+				SecurityGroups: []string{},
+				UserData:       "test",
+			},
+			shouldDrift: true,
+		},
+		{
+			launchTemplate: &ec2.LaunchTemplate{
+				LaunchTemplateName: aws.String("my-launch-config"),
+			},
+			latestVersion: &ec2.LaunchTemplateVersion{
+				LaunchTemplateData: &ec2.ResponseLaunchTemplateData{
+					IamInstanceProfile: &ec2.LaunchTemplateIamInstanceProfileSpecification{
```

Comment:
why do you always have to set the `IamInstanceProfile`?

Reply:
Test would get a NPE (in launchtemplate.go:204) otherwise because AWS structs are all pointers.
In this case it's OK to assume it will never be nil, because an IAM role is always added to the launchtemplate.

I will do more cleanup before merging to remove repeating code.

Reply:
I've got some fixes for those in https://github.com/eytan-avisror/instance-manager/pull/3